### PR TITLE
Add (nonfunctional) rulers and scrollbars to the viewport

### LINKED ISSUE #275: Viewport scrollbars and rulers

### PR DESCRIPTION
**Pull request by Keavon**
Closes #275 (and truly finalizes #112).

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/279)

### CODE REVIEW

**Review by AyeTbk — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/279)* status: all files reviewed, 1 unresolved discussion (waiting on @henryksloan and @Keavon)

___
*[client/web/src/components/widgets/rulers/CanvasRuler.vue, line 92 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/279#-Meq4886BKFQA98gYtmc:-Meq4886BKFQA98gYtmd:bh4lre9) ([raw file](https://github.com/graphiteeditor/graphite/blob/4a4ccebe1ff3940c8c5d6a2adefc1acebc164c1f/client/web/src/components/widgets/rulers/CanvasRuler.vue#L92)):*
> ```Vue
> 			if (roundedUp !== this.rulerLength) {
> 				this.rulerLength = roundedUp;
> 				this.svgBounds = isVertical ? { width: "16px", height: `${roundedUp}px` } : { width: `${roundedUp}px`, height: "16px" };
> ```

The width and height of 16px here, I imagine it's meant to reflect the RULER_THICKNESS? If so, should they be `${RULER_THICKNESS}px` instead?

File: client/web/src/components/widgets/rulers/CanvasRuler.vue [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			const isVertical = this.direction === RulerDirection.Vertical;
+			const lineDirection = isVertical ? "H" : "V";
+
+			let offsetStart = this.origin % this.majorMarkSpacing;
+			if (offsetStart < this.majorMarkSpacing) offsetStart -= this.majorMarkSpacing;
+
+			const divisions = this.majorMarkSpacing / this.mediumDivisions / this.minorDivisions;
+			const majorMarksFrequency = this.mediumDivisions * this.minorDivisions;
+
+			let d = "";
```
Comment (henryksloan) [inline client/web/src/components/widgets/rulers/CanvasRuler.vue:64]: I think `d` could have a more descriptive name

**Review by henryksloan — APPROVED** [review]
LGTM

File: client/web/src/components/widgets/rulers/CanvasRuler.vue (hunk 1 above)
Comment (AyeTbk) [reply client/web/src/components/widgets/rulers/CanvasRuler.vue:64]: I think 'd' here refers to the d attribute of svg elements. The d attribute defines a path to be drawn.
Not saying it couldn't be more descriptive, but I feel it is not too inapropriate.

Comment (henryksloan) [reply client/web/src/components/widgets/rulers/CanvasRuler.vue:64]: Ah, that make sense. Maybe a comment would be nice to clarify, but the name is probably acceptable. Alternatively, the variable could be named `path` or something, and a comment could indicate that it corresponds to the `d` attribute.

**Review by Keavon — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/279)* status: all files reviewed, 2 unresolved discussions (waiting on @AyeTbk and @henryksloan)

___
*[client/web/src/components/widgets/rulers/CanvasRuler.vue, line 64 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/279#-MenvZR_HJItCfpcddGU-r1-64:-MeqFoDY0ZeLWJXskdwK:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/4a4ccebe1ff3940c8c5d6a2adefc1acebc164c1f/client/web/src/components/widgets/rulers/CanvasRuler.vue#L64)):*
<details><summary><i>Previously, henryksloan (Henry Sloan) wrote…</i></summary><blockquote>

Ah, that make sense. Maybe a comment would be nice to clarify, but the name is probably acceptable. Alternatively, the variable could be named `path` or something, and a comment could indicate that it corresponds to the `d` attribute.
</blockquote></details>

Done.
___
*[client/web/src/components/widgets/rulers/CanvasRuler.vue, line 92 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/279#-Meq4886BKFQA98gYtmc:-MeqFpWr2BON_6HT27aX:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/4a4ccebe1ff3940c8c5d6a2adefc1acebc164c1f/client/web/src/components/widgets/rulers/CanvasRuler.vue#L92)):*
<details><summary><i>Previously, AyeTbk (Simon Desloges) wrote…</i></summary><blockquote>

The width and height of 16px here, I imagine it's meant to reflect the RULER_THICKNESS? If so, should they be `${RULER_THICKNESS}px` instead?
</blockquote></details>

Done.

**Review by AyeTbk — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/279)* status: all files reviewed, 1 unresolved discussion (waiting on @henryksloan)

**Review by Keavon — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/279)* status: all files reviewed, 1 unresolved discussion (waiting on @henryksloan)

___
*[client/web/src/components/widgets/rulers/CanvasRuler.vue, line 64 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/279#-MenvZR_HJItCfpcddGU-r1-64:-MeqjGFl5EBmXvwkWkRR:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/4a4ccebe1ff3940c8c5d6a2adefc1acebc164c1f/client/web/src/components/widgets/rulers/CanvasRuler.vue#L64)):*
<details><summary><i>Previously, Keavon (Keavon Chambers) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Pull request merged**